src/components/src/main/java/org/apache/jmeter/assertions/CompareAssertion.java

```diff
@@ -65,78 +65,82 @@ public AssertionResult getResult(SampleResult response) {
     }
 
     private void compareTime(CompareAssertionResult result) {
-        if (compareTime >= 0) {
-            long prevTime = -1;
-            SampleResult prevResult = null;
-            boolean success = true;
-            StringBuilder buf = new StringBuilder();
-            for(SampleResult sResult : responses) {
-                long currentTime = sResult.getTime();
-                if (prevTime != -1) {
-                    success = Math.abs(prevTime - currentTime) <= compareTime;
-                    prevResult = sResult;
-                }
-                if (!success) {
-                    result.setFailure(true);
-                    buf.setLength(0);
-                    appendResultDetails(buf, prevResult);
-                    buf.append(JMeterUtils.getResString("comparison_response_time")).append(prevTime); //$NON-NLS-1$
-                    result.addToBaseResult(buf.toString());
-                    buf.setLength(0);
-                    appendResultDetails(buf, sResult);
-                    buf.append(JMeterUtils.getResString("comparison_response_time")).append(currentTime); //$NON-NLS-1$
-                    result.addToSecondaryResult(buf.toString());
-                   result.setFailureMessage(
-                           JMeterUtils.getResString("comparison_differ_time")+ //$NON-NLS-1$
-                           compareTime+
-                           JMeterUtils.getResString("comparison_unit")); //$NON-NLS-1$
-                    break;
-                }
+        if (compareTime < 0) {
+            return;
+        }
+        long prevTime = -1;
+        SampleResult prevResult = null;
+        boolean success = true;
+        StringBuilder buf = new StringBuilder();
+        for (SampleResult sResult : responses) {
+            long currentTime = sResult.getTime();
+            if (prevTime != -1) {
+                success = Math.abs(prevTime - currentTime) <= compareTime;
+                prevResult = sResult;
```

Comment:
This looks strange since `prevResult` is used below, and it is especially strange that there's a second assignment below: `prevResult = sResult`

Reply:
Not sure what do about this one

Reply:
I think the `-1` is used as a guard here to check whether `prevTime` has been assigned already in a previous loop.

Reply:
@vlsi Ok, now I have understood, what you meant here. I think it is an error and will correct it with my next commit that should close this PR.